pipelines/build/common/riscv_version_out.groovy

```diff
@@ -0,0 +1,116 @@
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+/**
+ * This file is a jenkins job for extracting a version string from a cross-compiled, riscv binary.
+ * See https://github.com/AdoptOpenJDK/openjdk-build/issues/1773 for the inspiration for this.
+ *
+ * This file is referenced by the upstream job at pipelines/build/common/openjdk_build_pipeline.groovy.
+ * This job is run at build-scripts/job/utils/job/riscv-version-out/.
+ *
+ * The job looks like:
+ *  1. Switch into a riscv node
+ *  2. Retrieve the artifacts built in the upstream job
+ *  3. Run java -version and export to file
+ *  4. Expose to the upstream job by archiving file
+ */
+
+// TODO: ADD THE ACTIVE NODE TIMEOUT LOGIC HERE OR GET IT MERGED INTO JOB HELPER (https://github.com/AdoptOpenJDK/openjdk-build/issues/2235)
+
+node ("riscv&&ci.role.test") {
+    timestamps {
+        try {
+            Integer JOB_TIMEOUT = 1
+            timeout(time: JOB_TIMEOUT, unit: "HOURS") {
+                String jobName = params.UPSTREAM_JOB_NAME ? params.UPSTREAM_JOB_NAME : ""
+                String jobNumber = params.UPSTREAM_JOB_NUMBER ? params.UPSTREAM_JOB_NUMBER : ""
+                String jdkFileFilter = params.JDK_FILE_FILTER ? params.JDK_FILE_FILTER : ""
+                String fileName = params.FILENAME ? params.FILENAME : ""
+
+                println "[INFO] PARAMS:"
+                println "UPSTREAM_JOB_NAME = ${jobName}"
+                println "UPSTREAM_JOB_NUMBER = ${jobNumber}"
+                println "JDK_FILE_FILTER = ${jdkFileFilter}"
+                println "FILENAME = ${fileName}"
+
+                // Verify any previous binaries and versions have been cleaned out
+                if (fileExists('OpenJDKBinary')) {
+                    dir('OpenJDKBinary') {
+                        deleteDir()
+                    }
+                }
+
+                if (fileExists('RiscvVersionOuts')) {
+                    dir('RiscvVersionOuts') {
+                        deleteDir()
+                    }
+                }
+
+                String versionOut = ""
+
+                dir ("OpenJDKBinary") {
+                    // Retrieve built JDK & unzip
+                    println "[INFO] Retrieving build artifact from ${jobName}/${jobNumber} matching filter ${jdkFileFilter}"
+                    copyArtifacts(
+                        projectName: "${jobName}",
+                        selector: specific("${jobNumber}"),
+                        filter: "workspace/target/${jdkFileFilter}",
+                        fingerprintArtifacts: true,
+                        flatten: true
+                    )
+
+                    println "[INFO] Unzipping..."
+                    sh "tar -zxvf ${jdkFileFilter} && rm ${jdkFileFilter}"
+
+                    String jdkDir = sh(
+                        script: "ls | grep jdk*",
```

Comment:
`*` is superfluous here - suggest removing it to avoid potential confusion (Could also use `ls -d *jdk*`
```suggestion
                        script: "ls | grep jdk",
```